### PR TITLE
doc: add http2 docs

### PR DESCRIPTION
**Pull request by Claude-Zq**
#### What type of PR is this?
docs
#### What this PR does / why we need it (en: English/zh: Chinese):
en: Add documentation for HTTP2
zh: 添加了HTTP2的文档

#### Which issue(s) this PR fixes:
Fixes https://github.com/cloudwego/hertz/issues/465

### CODE REVIEW

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+---
+title: "HTTP2"
+date:  2022-12-10
+weight: 2
+description: >
+
+---
+HTTP2是HTTP协议的第二个版本，旨在通过改进第一个HTTP版本的许多缺点，使应用程序更快、更简单、更健壮。
```
Comment (Duslia) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:8]: HTTP2 isn't the second version of HTTP

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+Hertz 提供了 HTTP2 的支持，参考了 [net/http2](https://github.com/golang/net/tree/master/http2) 的实现。
+
```
Comment (Duslia) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:11]: Cloud you add the config doc of HTTP2? The server and client config is in https://github.com/hertz-contrib/http2/tree/main/config

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+用法示例：
```
Comment (Duslia) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:12]: You can emphasize that it it the h2c example.

**Review by li-jin-gou — CHANGES_REQUESTED** [review]
Please refer to https://github.com/cloudwego/cloudwego.github.io/pull/441 to refine pr.

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 1 above)
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md]: Sorry for the reference to the [wrong article](https://www.upwork.com/resources/what-is-http2)

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
+---
+title: "HTTP2"
+date:  2022-12-10
+weight: 2
+description: >
+
+---
+HTTP2是HTTP协议的第二个版本，旨在通过改进第一个HTTP版本的许多缺点，使应用程序更快、更简单、更健壮。
+
+Hertz 提供了 HTTP2 的支持，参考了 [net/http2](https://github.com/golang/net/tree/master/http2) 的实现。
```
[the remark this answers is not in the document]
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md]: got it

**Comment by Claude-Zq** [comment]
> Please refer to #441 to refine pr.

Sorry,I referred to  [Websocket documentation](https://www.cloudwego.io/zh/docs/hertz/tutorials/basic-feature/protocol/websocket/) earlier (also in the **[Protocol](https://www.cloudwego.io/docs/hertz/tutorials/basic-feature/protocol/)** directory),and didn't notice that the http2 configuration was more complex and needed to be written into the documentation.
🤣

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 2 above)
Comment (Duslia) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:158]: There is no need give an example for every config. Just like here: https://www.cloudwego.io/docs/hertz/reference/config/   . Add a table is enough

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 5]
```diff
@@ -0,0 +1,2520 @@
...
+	"github.com/cloudwego/hertz/pkg/app/client"
+	"github.com/cloudwego/hertz/pkg/app/server"
+	"github.com/cloudwego/hertz/pkg/network/standard"
+	"github.com/cloudwego/hertz/pkg/protocol"
+	"github.com/hertz-contrib/http2/config"
+	"github.com/hertz-contrib/http2/factory"
+)
+
+const (
+	keyPEM = `-----BEGIN RSA PRIVATE KEY-----
```
Comment (Duslia) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:41]: The specific key content can use `keyPEM = "your key PEM"` instead. There is no need to show a complete keyPEM

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 6]
```diff
@@ -0,0 +1,2520 @@
+---
+title: "HTTP2"
+date:  2022-12-12
+weight: 2
+description: >
+
+---
+HTTP/2是对HTTP "在线 "表达方式的一种替代。它并不是对协议的彻底重写；HTTP方法、状态码和语义都是一样的，而且应该可以使用与HTTP/1.x相同的API（可能会有一些小的补充）来表示协议。
+
+该协议的重点是性能；具体而言，最终用户感知的延迟、网络和服务器资源的使用。一个主要目标是允许使用从浏览器到网站的单一连接。
```
[the remark this answers is not in the document]
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:10]: I think we should best refer to the official [http2 introduction](https://http2.github.io/) written by the [IETF HTTP Working Group](https://httpwg.github.io/).

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 5 above)
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md]: done

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 7]
```diff
@@ -0,0 +1,1890 @@
...
+
+
+### 客户端
+
+
+
+| 配置                           | 默认值                   | 介绍                                                |
+|------------------------------|-----------------------|---------------------------------------------------|
+| `MaxHeaderListSize`          | `0`，指使用默认的限制（10MB）    | 指http2规范中的`SETTINGS_MAX_HEADER_LIST_SIZE`。        |
+| `AllowHTTP`                  | `false`               | 设置是否允许http。                                       |
```
Comment (Duslia) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:414]: Add: the h2c switch

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 8]
```diff
@@ -0,0 +1,1890 @@
...
+```
+
+
+
+## 配置
+
+### 服务端
+
+| 配置                 | 默认值     |
+|:-------------------|:--------|
```
Comment (Duslia) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:191]: You can add introduction for each config.

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 9]
```diff
@@ -0,0 +1,1890 @@
...
+函数签名：
+
+```go
+func WithReadTimeout(t time.Duration) Option
+```
+
+示例代码:
+
+```go
+package main
```
Comment (Duslia) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:209]: Delete example code for each option. It is too complex.

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 7 above)
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md]: done

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 8 above)
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md]: done

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 9 above)
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:198]: done and I have tried to show all the options in one example

**Review by Duslia — APPROVED** [review]

**Comment by li-jin-gou** [comment]
PATl @L2ncE @justlorain @BaiZe1998

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 10]
```diff
@@ -0,0 +1,637 @@
+---
+title: "HTTP2"
+date:  2022-12-12
+weight: 2
+description: >
+
+---
+HTTP/2是对HTTP "在线" 表达方式的一种替代。它并不是对协议的彻底重写；HTTP方法、状态码和语义都是一样的，而且应该可以使用与HTTP/1.x相同的API（可能会有一些小的补充）来表示协议。
+
+该协议的重点是性能；具体而言，最终用户感知的延迟、网络和服务器资源的使用。一个主要目标是允许使用从浏览器到网站的单一连接。
```
Comment (BaiZe1998) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:10]: "该协议的重点是性能；具体而言，最终用户感知的延迟、网络和服务器资源的使用..." => "该协议的侧重点是性能：缩短用户感知的延迟、减少网络和服务器资源的使用..."

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 11]
```diff
@@ -0,0 +1,637 @@
...
+
+	"github.com/cloudwego/hertz/pkg/app"
+	"github.com/cloudwego/hertz/pkg/app/client"
+	"github.com/cloudwego/hertz/pkg/app/server"
+	"github.com/cloudwego/hertz/pkg/network/standard"
+	"github.com/cloudwego/hertz/pkg/protocol"
+	"github.com/hertz-contrib/http2/config"
+	"github.com/hertz-contrib/http2/factory"
+)
+
```
Comment (BaiZe1998) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:39]: The sample code seems to require formatting.

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 12]
```diff
@@ -0,0 +1,637 @@
+---
+title: "HTTP2"
+date:  2022-12-12
+weight: 2
+description: >
+
+---
+HTTP/2是对HTTP "在线" 表达方式的一种替代。它并不是对协议的彻底重写；HTTP方法、状态码和语义都是一样的，而且应该可以使用与HTTP/1.x相同的API（可能会有一些小的补充）来表示协议。
```
Comment (justlorain) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:8]: Add a space between Chinese and English

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 13]
```diff
@@ -0,0 +1,637 @@
...
+		}
+		ctx.JSON(http.StatusOK, r)
+	})
+
+	go runClient()
+
+	h.Spin()
+}
+```
+
```
Comment (justlorain) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:291]: remove unnecessary blank lines

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 14]
```diff
@@ -0,0 +1,637 @@
...
+```go
+func WithReadTimeout(t time.Duration) Option
+```
+
+
+
+#### WithDisableKeepAlive
+
+用于设置是否禁用 `keep-alive`，默任不禁用。
```
Comment (justlorain) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:308]: "默任" -> "默认"

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 15]
```diff
@@ -0,0 +1,637 @@
...
+```
+
+
+
+
+#### WithMaxHeaderListSize
+
+用于设置 `SETTINGS_MAX_HEADER_LIST_SIZE`。
+
+与http2规范不同，这里的`0`表示使用默认限制（目前是10MB）。如果想表示无限，可以设置为一个尽可能大的值（`0xffffffff` 或 `1<<32-1`）。
```
Comment (justlorain) [inline content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:460]: http2 -> HTTP2

File: content/en/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 16]
```diff
@@ -0,0 +1,624 @@
...
+
+```go
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/cloudwego/hertz/pkg/app/client/retry"
```
Comment (justlorain) [inline content/en/docs/hertz/tutorials/basic-feature/protocol/http2.md:345]: import style

File: content/en/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 17]
```diff
@@ -0,0 +1,624 @@
...
+Function Signature:
+
+```go
+func WithStrictMaxConcurrentStreams(strictMaxConcurrentStreams bool) ClientOption
+```
+
+
+#### WithPingTimeout
+
+Sets the timeout for `Ping` responses, after which the connection will be closed if no response is received to `Ping `.The default is `15s`.
```
Comment (justlorain) [inline content/en/docs/hertz/tutorials/basic-feature/protocol/http2.md:514]: Sets -> Set

File: content/en/docs/hertz/tutorials/basic-feature/protocol/http2.md [hunk 18]
```diff
@@ -0,0 +1,624 @@
...
+Function Signature:
+
+```go
+func WithPingTimeout(pt time.Duration) ClientOption 
+```
+
+
+#### WithAllowHTTP
+
+Used to set whether to allow http. if enabled, the client will use h2c mode. Not enabled by default.
```
Comment (justlorain) [inline content/en/docs/hertz/tutorials/basic-feature/protocol/http2.md:525]: http -> HTTP

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 14 above)
Comment (li-jin-gou) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:308]: 😂

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 11 above)
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:38]: done

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 10 above)
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md]: done

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 12 above)
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md]: done

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 13 above)
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md:290]: done

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 14 above)
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md]: done

File: content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 15 above)
Comment (Claude-Zq) [reply content/zh/docs/hertz/tutorials/basic-feature/protocol/http2.md]: done

File: content/en/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 16 above)
Comment (Claude-Zq) [reply content/en/docs/hertz/tutorials/basic-feature/protocol/http2.md]: done

File: content/en/docs/hertz/tutorials/basic-feature/protocol/http2.md (hunk 18 above)
Comment (Claude-Zq) [reply content/en/docs/hertz/tutorials/basic-feature/protocol/http2.md]: done

**Review by Duslia — APPROVED** [review]

**Comment by li-jin-gou** [comment]
LGTM

**Pull request merged**